slsim/LsstSciencePipeline/util_lsst.py

```diff
@@ -210,3 +214,373 @@ def opsim_variable_lens_injection(
+            min_mjd, max_mjd, 1000
+        ):  # Test multiple candidates
+            start, end = (
+                zero_point_mjd + lightcurve_range[0],
+                zero_point_mjd + lightcurve_range[1],
+            )
+            points_in_range = np.sum((mjd_times >= start) & (mjd_times <= end))
+
+            if points_in_range >= min_points:
+                start_point = zero_point_mjd  # Return the first valid zero_point_mjd
+            else:
+                warning_msg = (
+                    "In the given observations, there is no observation window which contains"
+                    " %s observational points. So, random start point is choosen"
+                    % min_points
+                )
+                warnings.warn(warning_msg, category=UserWarning, stacklevel=2)
+                start_point = transient_event_time_mjd(min_mjd=min_mjd, max_mjd=max_mjd)
+    else:
+        start_point = transient_event_time_mjd(min_mjd=min_mjd, max_mjd=max_mjd)
+    return start_point
+
+
+def transient_data_with_cadence(
+    lens_class,
+    exposure_data,
+    transform_pix2angle=None,
+    num_pix=61,
+    optimized_cadence=True,
+    min_points=100,
+    noise=True,
+    symmetric=False,
+):
+    """Puts lensed transient into the provided cadence. For LSST, this will be
+    cadence from opsim.
+
+    :param lens_class: Lens() object
+    :param exposure_data: An astropy table of exposure data. One entry
+        of table_list_data generated from the
+        opsim_time_series_images_data function. It must contain the rms
+        of background noise fluctuations (column name should be
+        "bkg_noise"), psf kernel for each exposure (column name should
+        be "psf_kernel", these are pixel psf kernel for each single
+        exposure images in time series image), observation time (column
+        name should be "obs_time", these are observation time in days
+        for each single exposure images in time series images), exposure
+        time (column name should be "expo_time", these are exposure time
+        for each single exposure images in time series images),
+        magnitude zero point (column name should be "zero_point", these
+        are zero point magnitudes for each single exposure images in
+        time series image), coordinates of the object (column name
+        should be "calexp_center"), these are the coordinates in (ra,
+        dec), and the band in which the observation is taken (column
+        name should be "band").
+    :param transform_pix2angle: Transformation matrix (2x2) of pixels
+        into coordinate displacements.
+    :param num_pix: number of pixel for the image.
+    :param optimized_cadence: Boolean. If True, search for the time
+        interval where given minimum number of observation points occur.
+    :param min_points: minimum number of observations in a certain time
+        window.
+    :param noise: Boolean. If True, a gaussian noise is added to the
```

Comment:
Gaussian noise in flux, I would guess. Can you change it to poisson? (then it's easier to get upper limits)